src/index.ts

```diff
@@ -170,441 +155,196 @@ export class ERC725 {
    *
-   * await myERC725.getData('LSP3Profile');
-   * // > {
-   * //   hashFunction: 'keccak256(utf8)',
-   * //   hash: '0xd96ff7776660095f661d16010c4349aa7478a9129ce0670f771596a6ff2d864a',
-   * //   url: 'ipfs://QmbTmcbp8ZW23vkQrqkasMFqNg2z1iP4e3BCUMz9PKDsSV'
-   * // }
+   * ```javascript title="getData - all keys from schema"
+   * https://github.com/ERC725Alliance/erc725.js/blob/feature/simplify-api/examples/src/getData.js#L7-L30
    * ```
-   * :::note Try it
-   * https://stackblitz.com/edit/erc725js-get-data?devtoolsheight=66&file=index.js
-   * :::
-   */
-  async getData(key: string, customSchema?: Erc725Schema) {
-    if (!isAddress(this.options.address)) {
-      throw new Error('Missing ERC725 contract address.');
-    }
-    if (!this.options.provider) {
-      throw new Error('Missing provider.');
-    }
-
-    const schema = customSchema ? [customSchema] : this.options.schema;
-    const keySchema = getSchemaElement(schema, key);
-
-    // Get all the raw data possible.
-    const rawData = await this.options.provider.getData(
-      this.options.address,
-      keySchema.key,
-    );
-    // Decode and return the data
-
-    if (keySchema.keyType.toLowerCase() === 'array') {
-      const dat = [{ key: keySchema.key, value: rawData }];
-      const res = await this.getArrayValues(keySchema, dat);
-
-      // Handle empty arrays
-      if (res && res.length > 0) {
-        res.push(dat[0]); // add the raw data array length
-        return decodeKey(keySchema, res);
-      }
-
-      return []; // return empty array if no results
-    }
-
-    return decodeKey(keySchema, rawData);
-  }
-
-  /**
-   * Get all available data from the contract as per the class schema definition.
-   * @returns An object with schema element key names as properties, with corresponding associated decoded data as values.
    *
-   * **Example**
+   * ```javascript reference title="getData - one key  "
+   * https://github.com/ERC725Alliance/erc725.js/blob/feature/simplify-api/examples/src/getData.js#L32-L41
+   * ```
    *
-   * ```javascript
-   * await myERC725.getAllData();
-   * // >
-   * // {
-   * //     'SupportedStandards:ERC725Account': '0xafdeb5d6',
-   * //     LSP3Profile: {
-   * //       hashFunction: 'keccak256(utf8)',
-   * //       hash: '0x8700cccf72722106436cbc5309a8ebb308224d5f601990c070ea751a6bed4fc0',
-   * //       url: 'ipfs://QmV8K2ZPZHErvVzjSE7vewgEzvfLnhdea8RLJRqZGNu9Je'
-   * //     },
-   * //     LSP1UniversalReceiverDelegate: '0x1183790f29BE3cDfD0A102862fEA1a4a30b3AdAb',
-   * //     'LSP3IssuedAssets[]': [
-   * //       '0xD94353D9B005B3c0A9Da169b768a31C57844e490',
-   * //       '0xDaea594E385Fc724449E3118B2Db7E86dFBa1826',
-   * //       ...
-   * //     ]
-   * // }
+   * ```javascript reference title="getData - many keys"
+   * https://github.com/ERC725Alliance/erc725.js/blob/feature/simplify-api/examples/src/getData.js#L43-L56
```

Comment:
You are right, I'm aware of it. The plugin we are using for this (`docusaurus-theme-github-codeblock`) is okay, but I would rather have it inlining the code at build time (docusaurus) rather than downloading the code when displaying the page. I'll look into that at some point, in theory it should not be too hard to adapt the plugin.